### PR TITLE
Tests Completed for Section2

### PR DESCRIPTION
**Pull request by siva1982**

### CODE REVIEW

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature [hunk 1]
```diff
@@ -0,0 +1,176 @@
+@Regression
+Feature: Section2 Clearance Prelodged
+
+  Background:
+    Given I clear data in cache
+  #The initial section 2 Journey starts with below combination
+  #isEIDR=NO, #isExporter=NO, isExporterEORI=NO, holdTheContract=NO, carrierEoriNumber=No
+
+  @Section2 @ClearancePrelodgedJourney
```
Comment (dhanist-gadhiraju) [inline src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:9]: we were already using tags @ClrDec for clearance declaration can we use the same thing as it is shorter tag. also we will not run prelodged and arrived seperatly we can use one tag for clearance. I found similar thing across other spec.  for standard we are using @StdDec, @ClrDec, @OcaDec, @SimDec, @SupDec these tags are used across section1, section3, section 4

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    And I select first Authorisation code ACR and enter eori as GB123456789008
+    And I click continue
+    Then I should land on Authorisations-Required-List page
+    And I select No to add another authorisation
+    And I click continue
+    Then I should land on MiniCYA-Section-2 page
+    And I check the MiniCYA page for Section-2
+    And I click continue on MiniCya
+
+    #  Given the user is on the Entry into declarant records page
```
Comment (dhanist-gadhiraju) [inline src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:56]: Looking at this comment is look like we have commented out the steps. please can you remove given, when , then words and leave rest of the comments

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    Then I select Yes to is this an entry into declarant records
+    And I click continue
+    Then I should land on Person-Presenting-Goods page
+    And I enter entered eori as GB121212121212
+    And I click continue
+    Then I should land on Exporter-Eori-Number page
+    And I navigate to Summary Section2 page
+    And I check the MiniCYA page for Section-2
+
+    #  Given the user is on the Entry into declarant records page
```
Comment (dhanist-gadhiraju) [inline src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:72]: same as above

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    And I provide Representatives Eori as GB121012121212
+    And I click continue
+    Then I should land on Representation-Type-Agreed page
+    And I select Direct as the type of representation
+    And I click continue
+    Then I should land on Consignee-Details page
+    And I navigate to Summary Section2 page
+    And I check the MiniCYA page for Section-2
+
+    #  Given the user is on the Entry into declarant records page
```
Comment (dhanist-gadhiraju) [inline src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:95]: same as above

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/runner/Smoke.scala [hunk 5]
```diff
@@ -25,6 +25,6 @@ import org.junit.runner.RunWith
   glue = Array("uk.gov.hmrc.test.ui.cucumber.stepdefs"),
   plugin =
     Array("pretty", "html:target/cucumber.html", "json:target/cucumber.json", "junit:target/test-reports/Runner.xml"),
-  tags = "@Section1"
+  tags = "@ClearanceArrivedJourney"
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/runner/Smoke.scala:28]: update tag to smoke

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala [hunk 6]
```diff
@@ -51,6 +51,6 @@ object DetailKeys {
   // /enter-a-mucr
   val Mucr: DetailKey = DetailKey("MUCR", section1)
 
-  // /enterry-into-declarants-records
+  // /entery-into-declarants-records
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala:54]: spelling mistake

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DeclarantDetailsPage.scala [hunk 7]
```diff
@@ -19,11 +19,13 @@ package uk.gov.hmrc.test.ui.pages.section1
...
 import uk.gov.hmrc.test.ui.pages.base.Constants.{Clearance, Common, yesNo}
 import uk.gov.hmrc.test.ui.pages.base.TariffLinks.{declarantDetails, declarantDetailsCL}
+import uk.gov.hmrc.test.ui.pages.section1.DeclarationChoicePage.isClearance
 import uk.gov.hmrc.test.ui.pages.section1.DetailKeys.DeclarationEori
+import uk.gov.hmrc.test.ui.pages.section2.EntryIntoDeclarantRecordsPage
 
 object DeclarantDetailsPage extends BasePage {
 
-  def backButtonHref: String = DeclarationTypePage.path
+  def backButtonHref: String = if(isClearance)EntryIntoDeclarantRecordsPage.path else DeclarationTypePage.path
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DeclarantDetailsPage.scala:28]: Please add the blanks `if (isClearance) EntryIntoDeclarantRecordsPage.path else

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala (hunk 6 above)
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala:54]: `/entry`

Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala:54]: ya this is speliing mistake , so i added the correct spelling

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala [hunk 8]
```diff
@@ -19,17 +19,29 @@ package uk.gov.hmrc.test.ui.pages.section2
...
+import uk.gov.hmrc.test.ui.pages.section1.DetailKeys.AdditionalDeclarationType
 import uk.gov.hmrc.test.ui.pages.section2.DetailKeys.{NoAuthorisationRequired, ProcedureChoice}
 
 object AuthorisationsYesNoPage extends BasePage {
 
-  def backButtonHref: String = ProcedureChoicePage.path
+  def backButtonHref: String = if (isOccasional) OtherPartiesInvolvedListPage.path
+  else if(isClearance) ConsigneeDetailsPage.path
+  else ProcedureChoicePage.path
   val path: String = "/declaration/is-authorisation-required"
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:31]: One blank row on top of this row

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DeclarantDetailsPage.scala (hunk 7 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DeclarantDetailsPage.scala:28]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala [hunk 9]
```diff
@@ -19,17 +19,29 @@ package uk.gov.hmrc.test.ui.pages.section2
...
 
   def title: String =
-    detail(ProcedureChoice) match {
-      case "Permanent" => "Do you have any authorisations you want to declare?"
-      case "Temporary" => "Do you want to add any authorisations?"
+    if (isOccasional || isClearance) "Do you want to add any authorisations?"
+    else if(detail(AdditionalDeclarationType).contains("Pre-lodged")) {
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:35]: A blank after `if`

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala [hunk 10]
```diff
@@ -19,17 +19,29 @@ package uk.gov.hmrc.test.ui.pages.section2
...
+      detail(ProcedureChoice) match {
+        case "Permanent" => "Do you have any authorisations you want to declare?"
+        case "Temporary" => "Do you want to add any authorisations?"
+      }
+    } else{
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:40]: a blank after `else`

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ConsigneeDetailsPage.scala [hunk 11]
```diff
@@ -19,13 +19,26 @@ package uk.gov.hmrc.test.ui.pages.section2
...
-  def backButtonHref: String = maybeDetail(CarrierEORI) match {
-    case Some(_) => CarrierEORINumberPage.path
-    case None    => CarrierAddressPage.path
+  def backButtonHref: String = if (isSupplementary && !isExporter)
+    RepresentationTypeAgreedPage.path
+  else if (isSupplementary && isExporter)
+    AreYouTheExporterPage.path
+  else if (isThisExs)
+    ThirdPartyGoodsTransportationPage.path
+  else if(isClearance && !isExporter)
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ConsigneeDetailsPage.scala:35]: a blank after `if`

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ExporterEORINumberPage.scala [hunk 12]
```diff
@@ -19,10 +19,12 @@ package uk.gov.hmrc.test.ui.pages.section2
 import uk.gov.hmrc.test.ui.pages.base.Constants.yesNo
 import uk.gov.hmrc.test.ui.pages.base.{BasePage, Detail}
 import uk.gov.hmrc.test.ui.pages.section2.DetailKeys.{EORI, ExporterEORI}
+import uk.gov.hmrc.test.ui.pages.section2.EntryIntoDeclarantRecordsPage.isEidr
+import uk.gov.hmrc.test.ui.pages.section2.IsThisExsPage.isThisExs
 
 object ExporterEORINumberPage extends BasePage {
 
-  def backButtonHref: String = AreYouTheExporterPage.path
+      def backButtonHref: String = if(isThisExs || isEidr)PersonPresentingGoodsPage.path else AreYouTheExporterPage.path
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ExporterEORINumberPage.scala:27]: Please use the blanks

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/IsThisExsPage.scala [hunk 13]
```diff
@@ -19,11 +19,13 @@ package uk.gov.hmrc.test.ui.pages.section2
 import uk.gov.hmrc.test.ui.pages.base.Constants.{yes, yesNo}
 import uk.gov.hmrc.test.ui.pages.base.{BasePage, Detail}
 import uk.gov.hmrc.test.ui.pages.section2.DetailKeys.{ExporterEORI, ExporterYesNo, IsThisExs}
+import uk.gov.hmrc.test.ui.pages.section2.EntryIntoDeclarantRecordsPage.isEidr
 
 object IsThisExsPage extends BasePage {
 
   def backButtonHref: String =
     if (detail(ExporterYesNo) == yes) AreYouTheExporterPage.path
+    else if(isEidr) ExporterEORINumberPage.path
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/IsThisExsPage.scala:28]: A blank after `if`

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+  override val expanderHrefs: Map[String, Seq[String]] = Map(
+    Common -> List(thirdPartyGoodsTransportation1, thirdPartyGoodsTransportation2),
+    Clearance -> List(thirdPartyGoodsTransportationCL1, thirdPartyGoodsTransportationCL2)
+  )
+
+  // No  => fillPage(no)
+  // Yes => fillPage(yes)
+
+  override def fillPage(values: String*): Unit = {
+
```
Comment (udinobi) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala:45]: Do you need this blank row?

File: src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala [hunk 15]
```diff
@@ -111,8 +111,10 @@ trait DriverHelper {
     findElementById(refSelector).sendKeys(refText)
   }
 
-  def fillTextBoxById(elementId: String, text: String): Unit =
+  def fillTextBoxById(elementId: String, text: String): Unit = {
+    waitForId(elementId).clear()
     waitForId(elementId).sendKeys(text)
```
[the remark this answers is not in the document]
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala:116]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/runner/Smoke.scala (hunk 5 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/runner/Smoke.scala:28]: Done

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature (hunk 3 above)
Comment (siva1982) [reply src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:72]: done

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature (hunk 2 above)
Comment (siva1982) [reply src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:56]: Done

File: src/test/resources/features/Section2/Clearance/ClearanceJourney.feature (hunk 1 above)
Comment (siva1982) [reply src/test/resources/features/Section2/Clearance/ClearanceJourney.feature:9]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala (hunk 14 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala:45]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+import uk.gov.hmrc.test.ui.pages.section2.DetailKeys.{EntryIntoDeclarantsRecords, ExporterYesNo, ThirdPartyGoodsTransportation}
+
+object ThirdPartyGoodsTransportationPage extends BasePage {
+
+  def backButtonHref: String = if (detail(ExporterYesNo) == yes) AreYouTheExporterPage.path
+  else if (isClearance) {
+    if (detail(EntryIntoDeclarantsRecords).equals("Yes")) ConsignorEORINumberPage.path else ConsignorDetailsPage.path
+  }
+  else RepresentationTypeAgreedPage.path
+  val path: String = "/declaration/third-party-goods-transportation"
```
[the remark this answers is not in the document]
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala:32]: Removed

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala (hunk 10 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:40]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala (hunk 9 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:35]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala [hunk 17]
```diff
@@ -18,11 +18,14 @@ package uk.gov.hmrc.test.ui.pages.section2
...
+import uk.gov.hmrc.test.ui.pages.section1.DeclarationChoicePage.{isClearance, isOccasional}
 import uk.gov.hmrc.test.ui.pages.section2.DetailKeys.{AuthorisationTypeLabel, section2}
 
 object AuthorisationsListPage extends BasePage {
 
-  def backButtonHref: String = ProcedureChoicePage.path
+  def backButtonHref: String = if(isOccasional)OtherPartiesInvolvedListPage.path
+  else if(isClearance) ConsigneeDetailsPage.path
+  else ProcedureChoicePage.path
   val path: String = "/declaration/authorisations-required"
```
[the remark this answers is not in the document]
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala:29]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala (hunk 6 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala:54]: Done

Comment (udinobi) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section1/DetailKeys.scala:54]: But isn't `/entery` yet? Should not be `/entry`?

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala (hunk 17 above)
[the remark this answers is not in the document]
Comment (udinobi) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala:29]: I cannot see a blank row between L28 and L29 yet

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala (hunk 8 above)
Comment (udinobi) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:31]: I cannot see a blank row between L30 and L31 yet

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala (hunk 16 above)
[the remark this answers is not in the document]
Comment (udinobi) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala:32]: I cannot see a blank row between L31 and L32 yet

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala (hunk 14 above)
Comment (udinobi) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/ThirdPartyGoodsTransportationPage.scala:45]: I still see the blank row at L45

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala (hunk 17 above)
[the remark this answers is not in the document]
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsListPage.scala:30]: please check it now

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala (hunk 8 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsYesNoPage.scala:32]: Done

**Review by dhanist-gadhiraju — DISMISSED** [review]

**Review by udinobi — APPROVED** [review]

**Pull request merged**